src/main/java/org/gridsuite/study/server/service/shortcircuit/ShortCircuitService.java

```diff
@@ -229,31 +233,65 @@ public static ShortCircuitParametersEntity toEntity(ShortCircuitParameters param
     }
 
     public static ShortCircuitParameters copy(ShortCircuitParameters shortCircuitParameters) {
-        return newShortCircuitParameters(shortCircuitParameters.getStudyType(), shortCircuitParameters.getMinVoltageDropProportionalThreshold(), shortCircuitParameters.isWithFeederResult(), shortCircuitParameters.isWithLimitViolations(), shortCircuitParameters.isWithVoltageResult(), shortCircuitParameters.isWithFortescueResult());
+        return newShortCircuitParameters(shortCircuitParameters.getStudyType(), shortCircuitParameters.getMinVoltageDropProportionalThreshold(), shortCircuitParameters.isWithFeederResult(), shortCircuitParameters.isWithLimitViolations(), shortCircuitParameters.isWithVoltageResult(), shortCircuitParameters.isWithFortescueResult(), shortCircuitParameters.isWithLoads(), shortCircuitParameters.isWithShuntCompensators(), shortCircuitParameters.isWithVSCConverterStations(), shortCircuitParameters.isWithNeutralPosition(), shortCircuitParameters.getInitialVoltageProfileMode(), shortCircuitParameters.getVoltageRanges());
     }
 
-    private static ShortCircuitParameters newShortCircuitParameters(StudyType studyType, double minVoltageDropProportionalThreshold, boolean withFeederResult, boolean withLimitViolations, boolean withVoltageResult, boolean withFortescueResult) {
-        ShortCircuitParameters shortCircuitParametersCopy = new ShortCircuitParameters()
+    public static ShortCircuitParameters newShortCircuitParameters(StudyType studyType, double minVoltageDropProportionalThreshold, boolean withFeederResult, boolean withLimitViolations, boolean withVoltageResult, boolean withFortescueResult, boolean withLoads, boolean withShuntCompensators, boolean withVscConverterStations, boolean withNeutralPosition, InitialVoltageProfileMode initialVoltageProfileMode, List<VoltageRange> voltageRanges) {
+        return new ShortCircuitParameters()
                 .setStudyType(studyType)
                 .setMinVoltageDropProportionalThreshold(minVoltageDropProportionalThreshold)
                 .setWithFeederResult(withFeederResult)
                 .setWithLimitViolations(withLimitViolations)
                 .setWithVoltageResult(withVoltageResult)
-                .setWithFortescueResult(withFortescueResult);
-        return shortCircuitParametersCopy;
+                .setWithFortescueResult(withFortescueResult)
+                //add fields related to version 1.2
+                .setWithLoads(withLoads)
+                .setWithShuntCompensators(withShuntCompensators)
+                .setWithVSCConverterStations(withVscConverterStations)
+                .setWithNeutralPosition(withNeutralPosition)
+                .setInitialVoltageProfileMode(initialVoltageProfileMode)
+                // if the initialVoltageProfileMode is NOMINAL the voltageRanges won't be taken care of when serializing the ShortCircuitParameters
```

Comment:
You can make it more simple. Something like: "voltageRanges is not taken in account when initialVoltageProfileMode=NOMINAL"

Reply:
OK.